test/unit/Reflection/Adapter/ReflectionClassTest.php

```diff
@@ -78,7 +79,6 @@ public function methodExpectationProvider(): array
             ['getProperties', null, [$mockProperty], []],
             ['hasConstant', null, true, ['foo']],
             ['getConstant', null, 'a', ['foo']],
-            ['getConstants', null, ['a', 'b'], []],
```

Comment:
Even considering the new test additions, shouldn't these data-provider entries be kept? In theory, the changes you introduced are BC compliant, so this should work as before 🤔

Reply:
The previous implementation relied upon `Reflection\ReflectionClass::getConstants()` which returns `array<string, scalar|array<scalar>|null>`.  Since we are keeping the filter logic out of the BR API, I chose not to delegate to `Reflection\ReflectionClass::getConstants()` but instead to `Reflection\ReflectionClass::getReflectionConstants()` so that I could filter on the visibility with `Reflection\ReflectionClassConstant::getModifiers()`.

I agree that removing this line in the data provider feels wrong, but I think the trade-offs are okay.  I'm open to suggestions on alternatives.  Not changing this method's delegation to `ReflectionClass::getConstants()` would still require a call to `getReflectionConstants()` in order to compare each constant's visibility.